unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/FileSeriesImporter.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+//A class for importing a STL series from a dir
+public class FileSeriesImporter
+{
+    GameObject seriesGameObject = new GameObject();
+    Mesh mesh = new Mesh();
+    private string[] filePaths;
+    private string extension;
+    bool cancelConvertion = false;
+    bool equalTopology = false; 
+
+    public GameObject GetGameObject()
+    {
+        return seriesGameObject;
+    }
+
+    public Mesh GetMesh()
+    {
+        return mesh;
+    }
+    
+    // Object constructor, initiates file series import.
+    public FileSeriesImporter(string rootFolder)
+    {
+        filePaths = Directory.GetFiles(rootFolder + @"\");
+        GetFilename();
+        LoadFiles();
+    }
+
+    private void GetFilename()
+    {
+        string fileName = Path.GetFileNameWithoutExtension(filePaths[0]);
+        extension = Path.GetExtension(filePaths[0]);
+        int lastCharIndex = fileName.LastIndexOf("-");
+        if (lastCharIndex == -1)
+            seriesGameObject.name = fileName;
+        else
+            seriesGameObject.name = fileName.Substring(0, lastCharIndex);
+
+    }
+
+    //loads meshes from separate files into one GameObject
+    private void LoadFiles()
+    {
+        SkinnedMeshRenderer skinnedMesh = seriesGameObject.AddComponent<SkinnedMeshRenderer>();
+        FileImporter fileImporter = new FileImporter(extension);
+
+        //Configuring progress bar
+        float progressChunk = (float) 1 / filePaths.Length;
+        cancelConvertion = EditorUtility.DisplayCancelableProgressBar("Convert STL series to a .prefab", "Conversion in progress", 0);
+
+        bool firstMesh = true;
+        for (int i = 0; i < filePaths.Length; i++)

```

Comment:
Why not `foreach`?

Reply:
It was easier to change number of iterations while testing, I'll make it foreach now.